flask/app.py

```diff
@@ -1370,6 +1390,33 @@ def url_defaults(self, f):
         self.url_default_functions.setdefault(None, []).append(f)
         return f
 
+    def _find_error_handler(self, e):
+        """Finds a registered error handler for the request’s blueprint.
+        If neither blueprint nor App has a suitable handler registered, returns None
+        """
+        exc_class, code = self._get_exc_class_and_code(type(e))
+        
+        def find_superclass(d):
+            if not d:
+                return None
+            for superclass in exc_class.mro():
+                if superclass is BaseException:
+                    return None
+                handler = d.get(superclass)
+                if handler is not None:
+                    d[exc_class] = handler  # cache for next time exc_class is raised
```

Comment:
They might not invalidate themselves correctly when the programmer is testing a very recent change they made, or otherwise warp results. As @adambyrtek said, you could argue that the programmer should simply be *aware* of those caches (as is the case with e.g. browser caches -- I always use icognito mode to test e.g. redirecting behavior), but in our case, it's questionable whether this cache even gives us any performance benefits (which is clearly the case with browser caches).

Also: In my opinion, caches are a tempting (and usually bad) answer to CPU-bound performance problems: They are simply there for delaying the writing of a more performant algorithm.